source/server/http/admin.cc

```diff
@@ -790,48 +851,59 @@ bool AdminImpl::removeHandler(const std::string& prefix) {
   return false;
 }
 
-void HystrixHandler::updateHystrixRollingWindow(HystrixData* hystrix_data, Server::Instance& server) {
+void HystrixHandler::updateHystrixRollingWindow(HystrixData* hystrix_data,
+    Server::Instance& server) {
   hystrix_data->stats_->incCounter();
 
   for (const Stats::CounterSharedPtr& counter : server.stats().counters()) {
+    // we save all upstream_rq stats. could be more specific.
     if (counter->name().find("upstream_rq_") != std::string::npos) {
-      //std::cout << "counter name: " << counter->name() << ", counter value: " << counter->value() << std::endl;
       hystrix_data->stats_->pushNewValue(counter->name(), counter->value());
     }
   }
 }
 
-void HystrixHandler::prepareAndSendHystrixStream(HystrixData* hystrix_data, Server::Instance& server) {
+void HystrixHandler::prepareAndSendHystrixStream(HystrixData* hystrix_data,
+    Server::Instance& server) {
   updateHystrixRollingWindow(hystrix_data,server);
 
   std::stringstream ss;
 
   for (auto& cluster : server.clusterManager().clusters()) {
-  hystrix_data->stats_-> getHystrixClusterStats(ss, cluster.second.get().info()->name(),
-        cluster.second.get().info()->resourceManager(Upstream::ResourcePriority::Default).pendingRequests().max(),
-        cluster.second.get().prioritySet().hostSetsPerPriority().size());
+    hystrix_data->stats_-> getHystrixClusterStats(ss,
+        cluster.second.get().info()->name(),
+        cluster.second.get().info()->resourceManager(
+            Upstream::ResourcePriority::Default).pendingRequests().max(),
+            cluster.second.get().prioritySet().hostSetsPerPriority().size());
   }
 
   Buffer::OwnedImpl data;
   data.add(ss.str());
 
   // using write() since we are sending network level
+  // is there an alternative to the const_cast?
   (const_cast<Network::Connection*>((hystrix_data->callbacks_)->connection()))->write(data);
 
-  const auto ms = std::chrono::milliseconds(Stats::HYSTRIX_ROLLING_WINDOW_IN_MS/Stats::HYSTRIX_NUM_OF_BUCKETS);
+  // TODO: move this outside to fixed place?
```

Comment:
We should move move it to the header file as a const variable. Also relevant to the variable in ping function

Reply:
i tried and didn't work, maybe you can try?